proxy/nasshp/types.go

```diff
@@ -0,0 +1,19 @@
+package nasshp
+
+type Verdict int
+
+const (
+	// Can't decide either way. This is useful for chaning filters.
+	VerdictUnknown Verdict = iota
+	// Let the request in.
+	VerdictAllow
+	// Block the request.
+	VerdictDrop
+)
+
+func (v Verdict) MergeOnlyAcceptAllow(vv Verdict) Verdict {
+	if v != VerdictAllow || vv != VerdictAllow {
```

Comment:
`VerdictUnknown` should be "non voting".

Given the set of checks to perform, I would succeed only if there is at least one VerdictAllow, and no VerdictDrop.

In terms of logic, that would mean

1)  initializing verdict to Unknown (see comment above)
2) changing MergeOnlyAcceptAllow like below
3) checking that the result is VerdictAllow, like you are doing already

MergeOnlyAcceptAllow would look something like:
```
switch vv {
case VerdictDrop: return VerdictDrop // no doubt, we have to drop.
case VerdictUnknown: return v // whatever we decided before, is still valid.
case VerdictAllow:
    if v == VerdictUnknown {
      return VerdictAllow
    }
    return v // if there was a previous Drop determination, let's stick with that. If it's not Drop, it's Allow already.
}
```